src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -367,6 +393,28 @@ private static async
 
                 int lastRead;
 
+                // Read up to 3 bytes to see if it's the UTF-8 BOM
+                do
+                {
+                    // No need for checking for growth, the minimal rent sizes both guarantee it'll fit.
+                    Debug.Assert(rented.Length >= JsonConstants.Utf8Bom.Length);
+
+                    lastRead = await stream.ReadAsync(
+                        rented,
+                        written,
+                        utf8BomLength - written,
```

Comment:
Again, to confirm, this will never be negative, correct? Maybe `Debug.Assert`

Reply:
I'm not sure what that assert would be guarding against... if a Stream implementation returns a number from ReadAsync bigger than count then pretty much everything ever touching a Stream is broken.

The guard against that would be ReadAsync itself, which should be guarding against a negative count.